cpp/models/ide_secir/model.cpp

```diff
@@ -21,101 +21,118 @@
 #include "ide_secir/parameters.h"
 #include "infection_state.h"
 #include "memilio/config.h"
+#include "memilio/epidemiology/age_group.h"
 #include "memilio/utils/logging.h"
 #include "memilio/math/eigen.h"
 
 #include "vector"
+#include <algorithm>
 
 namespace mio
 {
 namespace isecir
 {
 
-Model::Model(TimeSeries<ScalarType>&& init, ScalarType N_init, ScalarType deaths, ScalarType total_confirmed_cases,
-             const ParameterSet& Parameterset_init)
-    : parameters{Parameterset_init}
+Model::Model(TimeSeries<ScalarType>&& init, std::vector<ScalarType> N_init, std::vector<ScalarType> deaths,
+             int num_agegroups, std::vector<ScalarType> total_confirmed_cases)
+    : parameters{Parameters(AgeGroup(num_agegroups))}
     , m_transitions{std::move(init)}
-    , m_populations{TimeSeries<ScalarType>(Eigen::Index(InfectionState::Count))}
+    , m_populations{TimeSeries<ScalarType>(Eigen::Index(InfectionState::Count) * num_agegroups)}
     , m_total_confirmed_cases{total_confirmed_cases}
     , m_N{N_init}
+    , m_num_agegroups{num_agegroups}
+
 {
     if (m_transitions.get_num_time_points() > 0) {
         // Add first time point in m_populations according to last time point in m_transitions which is where we start the simulation.
         m_populations.add_time_point<Eigen::VectorXd>(
-            m_transitions.get_last_time(), TimeSeries<ScalarType>::Vector::Constant((int)InfectionState::Count, 0.));
+            m_transitions.get_last_time(),
+            TimeSeries<ScalarType>::Vector::Constant((int)InfectionState::Count * m_num_agegroups, 0.));
     }
     else {
         // Initialize m_populations with zero as the first point of time if no data is provided for the transitions.
         // This can happen for example in the case of initialization with real data.
         m_populations.add_time_point<Eigen::VectorXd>(
-            0, TimeSeries<ScalarType>::Vector::Constant((int)InfectionState::Count, 0));
+            0, TimeSeries<ScalarType>::Vector::Constant((int)InfectionState::Count * m_num_agegroups, 0));
     }
 
     // Set deaths at simulation start time t0.
-    m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Dead)] = deaths;
+    for (int group = 0; group < m_num_agegroups; group++) {
+        int Di                             = get_state_flat_index(Eigen::Index(InfectionState::Dead), group);
+        m_populations[Eigen::Index(0)][Di] = deaths[static_cast<Eigen::Index>((size_t)group)];
```

Comment:
Do you need all these type conversions?
General question: get_state_flat_index returns an int. Does it make it easier to return Eigen::Index or size_t to avoid some type conversions? Or use one of these in the for loop?